src/test/suite/component/download.test.ts

```diff
@@ -0,0 +1,26 @@
+import * as assert from 'assert';
+import {toTempFile, once} from '../../../component/download';
+import {succeeded} from '../../../utils/errorable';
+import * as fs from 'fs';
+
+suite('Run Download Test Suite', () => {
+   test('it creates temp file from given source url', async () => {
+      var result = await toTempFile(
+         'https://github.com/Azure/aks-devx-tools/blob/main/README.md'
+      );
```

Comment:
We should maybe read the contents of the temp file as well and validate them

Reply:
added validation for the contents of the downloaded file.